tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,259 @@
+        ParameterBuilderConfig(
+            **my_evaluation_dependency_0_parameter_builder.to_json_dict()
+        ),
+        ParameterBuilderConfig(
+            **my_evaluation_dependency_1_parameter_builder.to_json_dict(),
+        ),
+    ]
+    my_dependent_parameter_builder: ParameterBuilder = DummyParameterBuilder(
+        name="my_dependent_parameter_builder",
+        evaluation_parameter_builder_configs=evaluation_parameter_builder_configs,
+        data_context=None,
+    )
+
+    domain = Domain(
+        domain_type=MetricDomainTypes.TABLE,
+        domain_kwargs=None,
+        rule_name="my_rule",
+    )
+
+    variables: Optional[ParameterContainer] = None
+
+    parameter_container = ParameterContainer(parameter_nodes=None)
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    my_dependent_parameter_builder.build_parameters(
+        domain=domain,
+        variables=variables,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert mock_parameter_builder__build_parameters.call_count == 3
+
+    parameter_node: ParameterNode = get_parameter_value_and_validate_return_type(
+        domain=domain,
+        parameter_reference=my_dependent_parameter_builder.json_serialized_fully_qualified_parameter_name,
+        expected_return_type=None,
+        variables=variables,
+        parameters=parameters,
+    )
+    assert parameter_node == "mock.Mock"
+
+
+@pytest.mark.unit
+@mock.patch("test_parameter_builder.DummyParameterBuilder._build_parameters")
+def test_parameter_builder_dependencies_evaluated_mixed(
+    mock_parameter_builder__build_parameters: mock.MagicMock,
+):
+    my_evaluation_dependency_0_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_0",
+            evaluation_parameter_builder_configs=None,
+            data_context=None,
+        )
+    )
+    my_evaluation_dependency_1_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_1",
+            evaluation_parameter_builder_configs=None,
+            data_context=None,
+        )
+    )
+
+    domain = Domain(
+        domain_type=MetricDomainTypes.TABLE,
+        domain_kwargs=None,
+        rule_name="my_rule",
+    )
+
+    variables: Optional[ParameterContainer] = None
+
+    parameter_container = ParameterContainer(parameter_nodes=None)
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    my_evaluation_dependency_0_parameter_builder.build_parameters(
+        domain=domain,
+        variables=variables,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert mock_parameter_builder__build_parameters.call_count == 1
```

Comment:
Is this just asserting that the call on L223 was made?

Reply:
@cdkini Yes -- specifically, it is asserting that the call to the implementation method `_build_parameters()` as made by calling the public interface method `build_parameters()`.  Thanks.

Reply:
@cdkini What we can do is attempt to create a truly `Dummy` Domain and Parameters objects -- but the real test is that the implementation method `._build_parameters` is called, and in the production code it requires those arguments.  We must get the system to be able to execute this implementation method and us to confirm it.  Thanks.